aptos-move/block-executor/src/executor.rs

```diff
@@ -137,61 +143,54 @@ where
+        let mut apply_updates = |output: &E::Output| {
+            // First, apply writes.
+            for (k, v) in output.resource_write_set().into_iter().chain(
+                output
+                    .aggregator_v1_write_set()
+                    .into_iter()
+                    .map(|(state_key, write_op)| (state_key, (write_op, None))),
+            ) {
+                if prev_modified_keys.remove(&k).is_none() {
+                    updates_outside = true;
                 }
+                versioned_cache
+                    .data()
+                    .write(k, idx_to_execute, incarnation, v);
+            }
 
-                for (k, v) in output.module_write_set().into_iter() {
-                    if prev_modified_keys.remove(&k).is_none() {
-                        updates_outside = true;
-                    }
-                    versioned_cache.modules().write(k, idx_to_execute, v);
+            for (k, v) in output.module_write_set().into_iter() {
+                if prev_modified_keys.remove(&k).is_none() {
+                    updates_outside = true;
                 }
+                versioned_cache.modules().write(k, idx_to_execute, v);
+            }
 
-                // Then, apply deltas.
-                for (k, d) in output.aggregator_v1_delta_set().into_iter() {
-                    if prev_modified_keys.remove(&k).is_none() {
-                        updates_outside = true;
-                    }
-                    versioned_cache.data().add_delta(k, idx_to_execute, d);
+            // Then, apply deltas.
+            for (k, d) in output.aggregator_v1_delta_set().into_iter() {
+                if prev_modified_keys.remove(&k).is_none() {
+                    updates_outside = true;
                 }
+                versioned_cache.data().add_delta(k, idx_to_execute, d);
+            }
 
-                for (id, change) in output.aggregator_v2_change_set().into_iter() {
-                    if !prev_modified_aggregators.remove(&id) {
-                        updates_outside = true;
-                    }
+            for (id, change) in output.aggregator_v2_change_set().into_iter() {
+                // if !prev_modified_aggregators.remove(&id) {
+                //     updates_outside = true;
+                // }
 
-                    // TODO: figure out if change should update updates_outside
-                    // versioned_cache.aggregators().
-                    match change {
-                        AggregatorChange::Create(value) => versioned_cache
-                            .aggregators()
-                            .create_aggregator(id, idx_to_execute, value),
-                        AggregatorChange::Apply(apply) => versioned_cache
-                            .aggregators()
-                            .record_apply(id, idx_to_execute, apply),
-                    }
-                }
-            };
+                // TODO: figure out if change should update updates_outside
```

Comment:
@igor-aptos found an issue a day later
We should remove the entries that were not written after re-execution, otherwise, they are marked as estimate and now actually could be used as a bypass, and would certainly be a correctness issue.